build-info-extractor-nuget/src/main/java/org/jfrog/build/extractor/nuget/drivers/ToolchainDriverBase.java

```diff
@@ -68,6 +72,10 @@ protected String buildNugetSourceUrl(ArtifactoryBaseClient client, String repo)
                 .setScheme(rtUrl.getProtocol())
                 .setHost(rtUrl.getHost())
                 .setPath(rtUrl.getPath() + ARTIFACTORY_NUGET_API + repo);
+        int port = rtUrl.getPort();
+        if (port != -1) {
+            sourceUrlBuilder.setPort(port);
+        }
```

Comment:
The -1 check is redundant. Apache's URI builder already handles negative ports.
For our Apache HTTP client version see: [setPort](https://github.com/apache/httpcomponents-client/blob/rel/v4.5.12/httpclient/src/main/java/org/apache/http/client/utils/URIBuilder.java#L281) and [build](https://github.com/apache/httpcomponents-client/blob/rel/v4.5.12/httpclient/src/main/java/org/apache/http/client/utils/URIBuilder.java#L171).
For the latest Apache HTTP client code see [setPort](https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java#L485) and [build](https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java#L297).
```suggestion
                .setPath(rtUrl.getPath() + ARTIFACTORY_NUGET_API + repo)
                .setPort(rtUrl.getPort());

Reply:
Wow, thanks!.